module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -6289,7 +6288,7 @@ zfs_freebsd_copy_file_range(struct vop_copy_file_range_args *ap)
 		goto unlock;
 
 	error = zfs_clone_range(VTOZ(invp), ap->a_inoffp, VTOZ(outvp),
-	    ap->a_outoffp, ap->a_lenp, ap->a_fsizetd->td_ucred);
+	    ap->a_outoffp, (uint64_t *)ap->a_lenp, ap->a_fsizetd->td_ucred);
```

Comment:
It can't be converted like this.  On 32-bit platform you'll get a memory corruption.  You need a separate variable.

Reply:
Do you mean something like this ?
```c
uint64_t len;

len = *ap->a_lenp;
error = zfs_clone_range(VTOZ(invp), ap->a_inoffp, VTOZ(outvp),
    ap->a_outoffp, &len, ap->a_fsizetd->td_ucred);
*ap->a_lenp = (size_t)len;
```

Reply:
No need to separate declaration and statement, I think.

```c
uint64_t len = *ap->a_lenp;
```